src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -728,17 +728,17 @@ private bool TryParseBindAttribute(BindEntry bindEntry, out string valueAttribut
 
         foreach (var attribute in componentTagHelper.BoundAttributes)
         {
-            if (string.Equals(valueAttributeName, attribute.Name))
+            if (string.Equals(valueAttributeName, attribute.Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
> FWIW, there's a private helper in `TagHelperMatchingConventions` that retrieves the correct `StringComparison` based on `BoundAttributeDescriptor.CaseSensitive`. If that we're made public, this could be something like this.

Yeah, sounds good, I will do that, thanks.

> Also, this algorithm walks forwards and compares every attribute name three times. That could be improved by walking backwards and using three bools to track when `valueAttribute`, `changeAttribute`, and `expressionAttribute` are set. When a bool is set, we can avoid that particular string comparison. When all three are set, we can break out of the loop.

I can do that as well, thanks.

Reply:
> Although I think bools aren't necessary - we can check whether the variables are null.

Oh yes, you're right. I had misread the code above and my eyes were seeing `changedAttributeName` and `changedAttribute` as the same.